core/src/main/java/org/elasticsearch/common/geo/GeoDistance.java

```diff
@@ -127,12 +130,27 @@ public FixedSourceDistance fixedSourceDistance(double sourceLatitude, double sou
         }
     };
 
+    /** Returns a GeoDistance object as read from the StreamInput. */
+    @Override
+    public GeoDistance readFrom(StreamInput in) throws IOException {
+        return GeoDistance.values()[in.readVInt()];
```

Comment:
Great, unit test already exists, I will combine the existing two classes in the following commit.